### PR TITLE
Punishments

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

File: punishments.js [hunk 1]
```diff
@@ -28,6 +30,30 @@ let nameLockedUsers = Punishments.nameLockedUsers = Object.create(null);
...
+					bannedIps[data[i]] = '#ipban';
+				}
+			}
+			Punishments.checkRangeBanned = Cidr.checker(rangebans);
+			resolve();
+		});
+	});
+};
+
+loadBanlist();
```
Comment (Zarel) [inline punishments.js]: This isn't at our leisure at all!

Set a timeout for 1 millisecond.

Comment (Zarel) [inline punishments.js]: Actually, `setImmediate` also works.

File: punishments.js [hunk 2]
```diff
@@ -28,6 +30,30 @@ let nameLockedUsers = Punishments.nameLockedUsers = Object.create(null);
 let lockedRanges = Punishments.lockedRanges = Object.create(null);
 let rangelockedUsers = Punishments.rangeLockedUsers = Object.create(null);
 
+// load ipbans at our leisure
+let loadBanlist = Punishments.loadBanlist = function () {
+	return new Promise(function (resolve, reject) {
+		fs.readFile(path.resolve(__dirname, 'config/ipbans.txt'), (err, data) => {
+			if (err) return reject();
```
Comment (Zarel) [inline punishments.js]: You should pass the error to `reject`.

File: punishments.js (hunk 1 above)
Comment (bumbadadabum) [inline punishments.js]: >This isn't at our leisure at all!

HEY I took the wording from the original code.

Comment (Zarel) [inline punishments.js]: Yes, the original code was at our leisure. You moved the code to a point where it wasn't done at our leisure at all.

**Pull request merged**